streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -150,10 +150,10 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
             } else /* we previously owned this task, and we don't have it anymore, or it has changed active/standby state */ {
                 final Set<TopicPartition> inputPartitions = task.inputPartitions();
                 try {
-                    task.closeClean();
                     changelogReader.remove(task.changelogPartitions());
+                    task.closeClean();
```

Comment:
Yup! I also found this bug and fixed in another PR, but great to see it confirmed in yours as well :)